explorer/common/trace_stream.h

```diff
@@ -26,8 +44,11 @@ namespace Carbon {
 class TraceStream {
```

Comment:
Yes it would be nice to have it.
Can we have a constructor like this one? This would also avoid directly calling the `set_allowed_phases` method.
```cpp
explicit TraceStream(std::vector<ProgramPhase> allowed_phases_list = {
                           ProgramPhase::Unknown}) {
    set_allowed_phases(allowed_phases_list);
}
  ```

Reply:
Sounds good to me, though that'll mean it's the caller's responsibility to make sure that `Unknown` is always in the list (if we want `Unknown` output to always be traced).